dowhy/gcm/stochastic_models.py

```diff
@@ -127,7 +126,9 @@ def find_suitable_continuous_distribution(
                 generated_samples = distribution.rvs(size=distribution_samples.shape[0], loc=loc, scale=scale, *arg)
 
                 # Check the KL divergence between the distribution of the given and fitted distribution.
-                divergence = estimate_kl_divergence_continuous(distribution_samples, generated_samples)
+                from dowhy.gcm.divergence import estimate_kl_divergence_continuous_knn
+
+                divergence = estimate_kl_divergence_continuous_knn(distribution_samples, generated_samples)
```

Comment:
I assume you've run into import cycles and that's why you had to do a local import. This is likely a smell. I'm sure this problem is related to [this change in PR 932](https://github.com/py-why/dowhy/pull/932/files#diff-62eae184e3560bffa3e7cbc57e06f30910651732d961cf7e34c1a032e3c541fd). `PredictionModel` is an interface and you should not tie it to ML implementations in the `ml` sub-package (which automatically happens as soon as it's in that folder). I think moving `prediction_model.py` one folder up will relax the situation.

Reply:
It is unfortunately a bit more complicated because the new KL divergence method now also relies on the auto module (for optional automatic selection of the classifier model), which in turn imports the `EmpiricalDistribution` from the `stochastic_models` module. We might need to move these implementations into separate files. 
However, I will take a look this in a follow-up.